core/src/main/java/com/rudderstack/android/sdk/core/util/Utils.java

```diff
@@ -95,7 +95,12 @@ public static String getDeviceId(Application application) {
     }
 
     public static Map<String, Object> convertToMap(Object obj) {
-        String json = RudderGson.serialize(obj);
+        String json;
+        if (obj instanceof String) {
+            json = (String) obj;
+        } else {
+            json = RudderGson.serialize(obj);
+        }
```

Comment:
Rather than repeating this, we can extract this into a utility function.

Reply:
agreed, update this now.